torch/_inductor/freezing.py

```diff
@@ -147,12 +147,30 @@ def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
         )
 
 
-@torch.utils._python_dispatch._disable_current_modes()
 def invalidate_eager_modules():
-    for mod in torch._guards.TracingContext.get().module_context.nn_modules.values():
-        if not isinstance(mod, torch.nn.Module):
-            continue
+    with torch.utils._python_dispatch._disable_current_modes():
+        for (
+            mod
+        ) in torch._guards.TracingContext.get().module_context.nn_modules.values():
+            if not isinstance(mod, torch.nn.Module):
+                continue
+
+            for attr_name, tensor in list(
+                itertools.chain(
+                    mod.named_parameters(recurse=False),
+                    mod.named_buffers(recurse=False),
+                )
+            ):
+                with torch._dispatch.python.no_python_dispatcher():
+                    e_t = ErasedTensor(tensor, attr_name, mod)
+                if isinstance(tensor, torch.nn.Parameter):
+                    e_t.requires_grad_(True)
+                    e_t._is_param = True  # type: ignore[attr-defined]
+                setattr(mod, attr_name, e_t)
+
 
+def discard_traced_gm_params(mod: torch.fx.GraphModule):
```

Comment:
What is the difference between this one and invalidate_eager_modules above?
Given how confused git diff is, they are litteratlly the same with an extra for loop in the top one. Sohuld we just have one call the other?

Reply:
Yeah, it does seem like it. When you ignore whitespace you get the "normal" diff lol